core/src/main/java/org/elasticsearch/script/Template.java

```diff
@@ -40,9 +40,12 @@
     public static final String DEFAULT_LANG = "mustache";
 
     private XContentType contentType;
-
-    public Template() {
-        super();
+    
+    public Template(StreamInput in) throws IOException {
+        super(in);
+        if (in.readBoolean()) {
+            this.contentType = XContentType.readFrom(in);
+        }
```

Comment:
I'd move the writeTo implementation under this constructor so that they can be on the same page. It helps so much with eyeballing them for errors!